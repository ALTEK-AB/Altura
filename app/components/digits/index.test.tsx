import {render} from '@testing-library/react-native';
import React from 'react';
import {View} from 'react-native';
import renderer from 'react-test-renderer';

import Digit from '../digit';

import Digits from '.';

describe('<Digits />', () => {
  test('should render a <View /> component', () => {
    const {getByTestId} = render(<Digits currentLength={0} />);
    const digitsComponent = getByTestId('digits-component');
    expect(digitsComponent).toBeDefined();
    expect(digitsComponent.type).toEqual('View');
  });

  test('should render 6 <Digit /> components', () => {
    const {getAllByTestId} = render(<Digits currentLength={0} />);
    const digitComponents = getAllByTestId('digit-component');
    expect(digitComponents).toHaveLength(6);
  });

  test('Digits renders correctly', () => {
    const tree = renderer.create(<Digit isSelected={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
