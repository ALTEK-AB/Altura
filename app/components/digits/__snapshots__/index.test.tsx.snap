// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Digits /> Digits renders correctly 1`] = `
<View
  style={
    [
      {
        "borderRadius": 17.5,
        "height": 35,
        "width": 35,
      },
      {
        "backgroundColor": "#E2E8F6",
      },
      {
        "borderColor": "#2D3343",
        "borderWidth": 0.5,
      },
    ]
  }
  testID="digit-component"
/>
`;
